php/ui/component/class-system.php

```diff
@@ -0,0 +1,198 @@
+
+use Cloudinary\Support;
+
+/**
+ * System report Component.
+ *
+ * @package Cloudinary\UI
+ */
+class System extends Panel {
+
+	/**
+	 * Holds the components build blueprint.
+	 *
+	 * @var string
+	 */
+	protected $blueprint = 'report|settings/|input/';
+
+	/**
+	 * Holds the report data.
+	 *
+	 * @var array
+	 */
+	protected $report_data = array();
+
+	/**
+	 * Filter the input parts structure.
+	 *
+	 * @param array $struct The array structure.
+	 *
+	 * @return array
+	 */
+	protected function input( $struct ) {
+		$struct['element']             = 'input';
+		$struct['attributes']['type']  = 'hidden';
+		$struct['attributes']['name']  = 'system_report';
+		$struct['attributes']['value'] = wp_json_encode( $this->report_data );
+
+		return $struct;
+	}
+
+	/**
+	 * Filter the report parts structure.
+	 *
+	 * @param array $struct The array structure.
+	 *
+	 * @return array
+	 */
+	protected function report( $struct ) {
+
+		$struct['element'] = null;
+		// Add theme.
+		$this->theme();
+		// Add plugins.
+		$this->plugins();
+		// Add posts.
+		$this->posts();
+		// Add config.
+		$this->config();
+
+		return $struct;
+	}
+
+	/**
+	 * Build the posts report.
+	 */
+	protected function posts() {
+
+		$report_items = get_option( Support::REPORT_KEY, array() );
+		$report_items = array_unique( $report_items );
+		if ( ! empty( $report_items ) ) {
+			$post_data  = array();
+			$media_data = array();
+			foreach ( $report_items as $post_id ) {
+				$post_type = get_post_type( $post_id );
+				if ( 'attachment' === $post_type ) {
+					$data                   = wp_get_attachment_metadata( $post_id );
+					$data['all_meta']       = get_post_meta( $post_id );
+					$media_data[ $post_id ] = $data;
+				} else {
+					$data                  = get_post( $post_id, ARRAY_A );
+					$data['post_meta']     = get_post_meta( $post_id );
+					$post_data[ $post_id ] = $data;
+				}
+			}
+			if ( ! empty( $media_data ) ) {
+				$this->add_report_block( __( 'Media', 'cloudinary' ), 'media_report', $media_data );
+			}
+			if ( ! empty( $post_data ) ) {
+				$this->add_report_block( __( 'Posts', 'cloudinary' ), 'post_report', $post_data );
+			}
+		}
+	}
+
+	/**
+	 * Build the theme report.
+	 */
+	protected function theme() {
+		$active_theme = wp_get_theme();
+		$theme_data   = array(
+			'name'        => $active_theme->get( 'Name' ),
+			'version'     => $active_theme->get( 'Version' ),
+			'author'      => $active_theme->get( 'Author' ),
+			'author_url'  => $active_theme->get( 'AuthorURI' ),
+			'child_theme' => is_child_theme(),
+		);
+		$this->add_report_block( __( 'Theme', 'cloudinary' ), 'theme_status', $theme_data );
+
+	}
+
+	/**
+	 * Build the config report.
+	 */
+	protected function config() {
+		$struct['element'] = null;
+
+		$config = $this->setting->get_root_setting()->get_value();
+		unset( $config['connect'] );
+		$config['gallery']['gallery_config'] = json_decode( $config['gallery']['gallery_config'], JSON_PRETTY_PRINT );
```

Comment:
@DavidCramer

We need to factor in now the changes we are adding in the other work.